src/java.base/share/classes/jdk/internal/foreign/abi/SharedUtils.java

```diff
@@ -164,6 +165,27 @@ public static MethodHandle adaptUpcallForIMR(MethodHandle target, boolean dropRe
         return target;
     }
 
+    public static UpcallStubFactory arrangeUpcallHelper(MethodType targetType, boolean isInMemoryReturn, boolean dropReturn,
+                                                        ABIDescriptor abi, CallingSequence callingSequence) {
+        if (isInMemoryReturn) {
+            // simulate the adaptation to get the type
+            MethodHandle fakeTarget = MethodHandles.empty(targetType);
+            targetType = adaptUpcallForIMR(fakeTarget, dropReturn).type();
+        }
+
+        UpcallStubFactory factory = UpcallLinker.makeFactory(targetType, abi, callingSequence);
```

Comment:
Removed the API invocation not available.
Updated the file list https://github.com/ibmruntimes/openj9-openjdk-jdk/pull/572#issue-1634743873
FYI @ChengJin01

Reply:
Perhaps it would make more sense to add the new method in https://github.com/eclipse-openj9/openj9/pull/16985 (perhaps throwing an exception is sufficient)?

Reply:
The signature of the method `SharedUtils.arrangeUpcallHelper(...)` is totally specific to OpenJDK which is irrelevant to our implementation in OpenJ9. That being said, this method might need to be totally replaced if required to accommodate our code in OpenJ9.

Reply:
Added a stub method at https://github.com/eclipse-openj9/openj9/pull/16985.
OpenJDK change is restored.

Reply:
@ChengJin01 just suggested that `arrangeUpcallHelper` can be empty, and return `null`.

Reply:
I still believe the better approach is to add an implementation in https://github.com/eclipse-openj9/openj9/pull/16985 that just throws an exception. The symptoms will be much clearer if that method is used.

Reply:
It will end up with compilation failures as I disabled the all FFI code for JDK21 via https://github.com/eclipse-openj9/openj9/pull/17013 for the moment. In addition,  it is still unclear so far whether `UpcallLinker.makeFactory` will be used in OpenJ9 plus the arguments in the signature (e.g. `abi`, `callingSequence`) has nothing to do with the implementation in OpenJ9.

Reply:
I suggest that https://github.com/eclipse-openj9/openj9/pull/17013 should only disable the tests and not make any JCL changes.

Reply:
The method `SharedUtils.arrangeUpcallHelper()` are invoked by OpenJDK `CallArranger.arrangeUpcall()` for different platforms, all these invocations have been removed for OpenJ9, i.e., no `SharedUtils.arrangeUpcallHelper()` invocation.
I think no need to add an OpenJ9 implementation throwing an exception.

Reply:
Just talked to @keithc-ca offline, I will enable the JCL changes in https://github.com/eclipse-openj9/openj9/pull/17013 given there is a pretty small change in there (note: we will look into whether the method is used or updated to accommodate the existing code in OpenJ9).

Reply:
Added OpenJ9 `UpcallLinker.makeFactory()` at https://github.com/eclipse-openj9/openj9/pull/16985 
Restored `SharedUtils.arrangeUpcallHelper()`.